src/DotLiquid/ShopifyFilters.cs

```diff
@@ -0,0 +1,97 @@
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace DotLiquid
+{
+    /// <summary>
+    /// Shopify filters implemented by DotLiquid
+    /// </summary>
+    /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters"/>
+    public static class ShopifyFilters
+    {
+        /// <summary>
+        /// Converts a string into an MD5 hash.
+        /// </summary>
+        /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters/string-filters#md5"/>
+        /// <param name="input"/>
+        public static string Md5(string input)
+        {
+            return input.IsNullOrWhiteSpace() ? input : ComputeHash(MD5.Create(), input);
+        }
+
+        /// <summary> 
+        /// Converts a string into a SHA-1 hash.
+        /// </summary>
+        /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters/string-filters#sha1" />
+        /// <param name="input"></param>
+        public static string Sha1(string input)
+        {
+            return input.IsNullOrWhiteSpace() ? input : ComputeHash(SHA1.Create(), input);
+        }
+
+        /// <summary>
+        /// Converts a string into a SHA-256 hash.
+        /// </summary>
+        /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters/string-filters#sha256" />
+        /// <param name="input"></param>
+        public static string Sha256(string input)
+        {
+            return input.IsNullOrWhiteSpace() ? input : ComputeHash(SHA256.Create(), input);
+        }
+
+        /// <summary>
+        /// Converts a string into a SHA-1 hash using a hash message authentication code (HMAC).
+        /// Pass the secret key for the message as a parameter to the filter.
+        /// </summary>
+        /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters/string-filters#hmac_sha1" />
+        /// <param name="input" />
+        /// <param name="secretKey" />
+        public static string HmacSha1(string input, string secretKey)
+        {
+            if (input.IsNullOrWhiteSpace() || secretKey is null)
+                return input;
+
+            var hmac = new HMACSHA1(Encoding.UTF8.GetBytes(secretKey));
```

Comment:
This class implements dispose interface. You need a using block everywhere you create an instance.

Reply:
Fixed